extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/BridgeAbstractDiscoveryService.java

```diff
@@ -0,0 +1,95 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.digitalstrom.internal.discovery;
+
+import java.net.HttpURLConnection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.binding.digitalstrom.DigitalSTROMBindingConstants;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.config.Config;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.serverConnection.DsAPI;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.serverConnection.impl.DsAPIImpl;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingUID;
+
+/**
+ * The {@link DssBridgeAbstractDiscoveryParticipant} is responsible for discovering new and
+ * removed digitalSTROM bridges. It uses the central {@link AbstractDiscoveryService}.
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Matthias Siegele - Initial contribution
+ *
+ */
+public class BridgeAbstractDiscoveryService extends AbstractDiscoveryService {
+
+    /**
+     * Creates a new {@link DssBridgeAbstractDiscoveryParticipant}.
+     */
+    public BridgeAbstractDiscoveryService() {
+        super(DigitalSTROMBindingConstants.SUPPORTED_THING_TYPES_UIDS, 10);
+    }
+
+    private String hostAdress = "dss.local.";
+    private String dsid = null;
+
+    private void createResult() {
+        ThingUID uid = getThingUID();
+
+        Map<String, Object> properties = new HashMap<>(2);
+        properties.put(DigitalSTROMBindingConstants.HOST, hostAdress);
+        properties.put(DigitalSTROMBindingConstants.DS_ID, dsid);
+        if (uid != null) {
+            DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                    .withLabel("DigitalSTROM-Server").build();
+            thingDiscovered(result);
+        }
+    }
+
+    private ThingUID getThingUID() {
+        DsAPI digitalSTROMClient = new DsAPIImpl(hostAdress, Config.DEFAULT_CONNECTION_TIMEOUT,
+                Config.DEFAULT_READ_TIMEOUT);
+        int requestCode = digitalSTROMClient.checkConnection("123");
+        switch (requestCode) {
+            case HttpURLConnection.HTTP_OK:
+                dsid = digitalSTROMClient.getDSID("123");
+                if (dsid != null) {
+                    return new ThingUID(DigitalSTROMBindingConstants.THING_TYPE_DSS_BRIDGE, dsid);
+                }
+                break;
+            case HttpURLConnection.HTTP_UNAUTHORIZED:
+            case HttpURLConnection.HTTP_FORBIDDEN:
+            case -3:
+            case -4:
+            case -1:
+                return new ThingUID(DigitalSTROMBindingConstants.THING_TYPE_DSS_BRIDGE, "1");
+        }
+        return null;
+    }
+
+    @Override
+    protected void startScan() {
+        createResult();
```

Comment:
better schedule this using `scheduler.execute()`, so that this method does not block the further execution.